lsp-headerline.el

```diff
@@ -92,12 +92,12 @@ for caching purposes.")
   (let ((filename (f-filename file-path))
         (file-ext (f-ext file-path)))
     (if file-ext
-        (when (require 'lsp-treemacs nil t)
-          (if-let (icon (lsp-treemacs-get-icon file-ext))
-              (format "%s %s"
-                      (lsp-headerline--fix-image-background icon)
-                      filename)
-            filename))
+        (if (require 'lsp-treemacs nil t)
```

Comment:
Since the `else` clauses here are all the same, you could try combining some of the `if`s to cut down on the nesting. One way of doing it might be

``` emacs_lisp
(let ((filename (f-filename file-path)))
    (if (require 'lsp-treemacs nil t)
        (if-let* ((file-ext (f-ext file-path))
                  (icon (lsp-treemacs-get-icon file-ext)))
            (format "%s %s"
                    (lsp-headerline--fix-image-background icon)
                    filename)
          filename)
      filename))
```

There might be something better though.

Reply:
done